app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -133,7 +128,12 @@ public Void call() throws Exception {
                         Stream stream = subscriptions.get(i);
                         stream.getParsedColor();
                         stream.setSubscribed(true);
-                        streamDao.createOrUpdate(stream);
+                        try {
+                            streamDao.createOrUpdate(stream);
+                        }
+                        catch(Exception e) {
```

Comment:
when possible this should narrow to the most specific exception type. this goes for all caught Exceptions listed in this PR.